include/mamba/transaction_context.hpp

```diff
@@ -25,14 +26,17 @@ namespace mamba
     {
     public:
         TransactionContext() = default;
-        TransactionContext(const fs::path& prefix, const std::string& py_version);
+        TransactionContext(Context& ctx, const fs::path& prefix, const std::string& py_version);
 
         bool has_python;
         fs::path target_prefix;
         fs::path site_packages_path;
         fs::path python_path;
         std::string python_version;
         std::string short_python_version;
+        bool allow_softlinks = false;
```

Comment:
is this used?

Reply:
Yes it is:

When a hardlink fails, we have this fallback :
```
softlink = m_context->allow_softlinks;
copy = !softlink;
```